src/test/java/com/ericsson/eiffel/remrem/generate/service/EiffelRemremControllerUnitTest.java

```diff
@@ -0,0 +1,111 @@
+public class EiffelRemremControllerUnitTest {
+    
+    @InjectMocks
+    RemremGenerateController unit = new RemremGenerateController();
+    
+    @Mock
+    MsgService service;
+    
+    @Mock
+    MsgService service2;
+    
+
+    @Spy
+    private List<MsgService> msgServices = new ArrayList<MsgService>();
+    static String testInputName = "samplesemantics.json";
+    static JsonElement body;
+
+
+    @SuppressWarnings("resource")
+    @Before
+    public void setUp() throws Exception {
+        URL url = getClass().getClassLoader().getResource(testInputName);
+        String path = url.getPath().replace("%20"," ");
+        File file = new File(path);
+        body = new JsonParser().parse(new Scanner(file)
+            .useDelimiter("\\A").next());
+        MockitoAnnotations.initMocks(this);
+        msgServices.add(service);
+        msgServices.add(service2);
+        Mockito.when(service.getServiceName()).thenReturn("eiffelsemantics");
+        Mockito.when(service2.getServiceName()).thenReturn("eiffel3");
+        
+        Mockito.when(service.generateMsg(
+            Mockito.eq("eiffelactivityfinished"),
+            Mockito.anyObject()
+        )).thenReturn("{ \"result\":\"SUCCESS\" }");
+        
+        Mockito.when(service.generateMsg(
+                Mockito.eq("eiffeljobfinished"),
+                Mockito.anyObject()
+        )).thenReturn("{ \"result\":\"FAILURE\" }");
+        
+        Mockito.when(service2.generateMsg(
+                Mockito.eq("eiffeljobfinished"),
+                Mockito.anyObject()
+        )).thenReturn("{ \"result\":\"SUCCESS\" }");
+        
+        
+        Mockito.when(service2.generateMsg(
+                Mockito.eq("eiffelactivityfinished"),
+                Mockito.anyObject()
+        )).thenReturn("{ \"result\":\"FAILURE\" }");
+        
+        
+    }
+    
+    @Test 
+    public void testSemanticsEvent() throws Exception {        
+        JsonElement elem = unit.generate("eiffelsemantics", "eiffelactivityfinished", body.getAsJsonObject());
+        assertEquals(elem.getAsJsonObject().get("result").getAsString(), "SUCCESS");
+    }
+    
+    @Test
+    public void testSemanticsFailureEvent() throws Exception{
+        JsonElement elem = unit.generate("eiffelsemantics", "eiffeljobfinished", body.getAsJsonObject());
+        assertEquals(elem.getAsJsonObject().get("result").getAsString(), "FAILURE");
+    }
+    
+    @Test 
+    public void testEiffel3Event() throws Exception {        
+        JsonElement elem = unit.generate("eiffel3", "eiffeljobfinished", body.getAsJsonObject());
+        assertEquals(elem.getAsJsonObject().get("result").getAsString(), "SUCCESS");
+    }
+    
+    @Test
+    public void testEiffel3FailureEvent() throws Exception{
+        JsonElement elem = unit.generate("eiffel3", "eiffelactivityfinished", body.getAsJsonObject());
+        assertEquals(elem.getAsJsonObject().get("result").getAsString(), "FAILURE");
+    }   
+}
```

Comment:
Would be good with a unit test using a service that do not exist. unit.generate("non_existing_protocol",...